lib/auth/auth_with_roles.go

```diff
@@ -2625,8 +2625,14 @@ func (a *ServerWithRoles) UpsertKubeService(ctx context.Context, s services.Serv
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	_, isService := a.context.Checker.(BuiltinRoleSet)
```

Comment:
Can only kube service do this? There's `hasBuiltinRole` if you wanted to check for only kube service here.

Reply:
Kube or Proxy.
We already have this enforced in e.g. https://github.com/gravitational/teleport/blob/master/lib/auth/permissions.go#L547, I'd rather not duplicate that authz logic here.